src/blueprint/from_config/operators/expr.rs

```diff
@@ -0,0 +1,66 @@
+use crate::blueprint::*;
+use crate::config;
+use crate::config::{Config, ExprBody, Field};
+use crate::lambda::Expression;
+use crate::try_fold::TryFold;
+use crate::valid::Valid;
+
+struct CompilationContext<'a> {
+  config_field: &'a config::Field,
+  operation_type: &'a config::GraphQLOperationType,
+  config: &'a config::Config,
+}
+
+struct CompileIf<'a> {
+  context: &'a CompilationContext<'a>,
+  cond: Box<ExprBody>,
+  then: Box<ExprBody>,
+  els: Box<ExprBody>,
+}
+
+fn compile_if(input: CompileIf) -> Valid<Expression, String> {
+  let context = input.context;
+  let cond = input.cond;
+  let then = input.then;
+  let els = input.els;
+
+  compile(context, *cond)
+    .trace("cond")
+    .map(Box::new)
+    .zip(compile(context, *then).trace("then").map(Box::new))
+    .zip(compile(context, *els).trace("else").map(Box::new))
+    .map(|((cond, then), els)| Expression::If { cond, then, els })
+}
+
```

Comment:
do you mind putting this below `compile` fn? it'll improve readability as we usually read code top-to-bottom.